modules/basicfuncs/tests/test_basicfuncs.c

```diff
@@ -219,6 +219,19 @@ Test(basicfuncs, test_numeric_funcs)
   assert_template_format("$(% 10000000000 5000000001)", "4999999999");
   assert_template_format("$(* 5000000000 2)", "10000000000");
   assert_template_format("$(- 10000000000 5000000000)", "5000000000");
+
+  assert_template_format("$(+ 1.5 .25)", "1.750000");
+  assert_template_format("$(- -1.5 .25)", "-1.750000");
+  assert_template_format("$(/ 3 2)", "1");
+  assert_template_format("$(/ 3.0 2)", "1.500000");
+  assert_template_format("$(/ 3 2.0)", "1.500000");
+  assert_template_format("$(* 1.5 2.0)", "3.000000");
+  assert_template_format("$(% 3.14 0.7)", "0.340000");
+
+  assert_template_format("$(round 2.0)", "2");
+  assert_template_format("$(round 0.5)", "1");
+  assert_template_format("$(floor 0.7)", "0");
+  assert_template_format("$(ceil 0.2)", "1");
```

Comment:
[optional]
Could you please extend the UT test set with scientific notation number parsing cases. Like `$(+ 1e-1 0)` (it already supports those).

Reply:
Sure, will use `5e-1` because, you know IEEE-754 cannot represent a lot of numbers with 20 digits precision: actual=0.10000000000000000555, expected=0.10000000000000000000 :smile: